src/test/app/MultiSign_test.cpp

```diff
@@ -1134,6 +1134,80 @@ class MultiSign_test : public beast::unit_test::suite
         env.close();
     }
 
+    void test_signForHash()
+    {
+        // Make sure that the "hash" field returned by the "sign_for" RPC
+        // command matches the hash returned when that command is sent
+        // through "submit_multisigned".  Make sure that hash also locates
+        // the transaction in the ledger.
+        using namespace jtx;
+        Account const alice {"alice", KeyType::ed25519};
+
+        Env env(*this);
+        env.fund (XRP(1000), alice);
+        env.close();
+
+        env (signers (alice, 2, {{bogie, 1}, {ghost, 1}}));
+        env.close();
+
+        // Use sign_for to sign a transaction where alice pays 10 XRP to
+        // masterpassphrase.
+        std::uint32_t baseFee = env.current()->fees().base;
+        Json::Value jvSig1;
+        jvSig1[jss::account] = bogie.human();
+        jvSig1[jss::secret]  = bogie.name();
+        jvSig1[jss::tx_json][jss::Account]         = alice.human();
+        jvSig1[jss::tx_json][jss::Amount]          = 10000000;
+        jvSig1[jss::tx_json][jss::Destination]     = env.master.human();
+        jvSig1[jss::tx_json][jss::Fee]             = 3 * baseFee;
+        jvSig1[jss::tx_json][jss::Sequence]        = env.seq(alice);
+        jvSig1[jss::tx_json][jss::TransactionType] = "Payment";
+
+        Json::Value jvSig2 = env.rpc (
+            "json", "sign_for", to_string (jvSig1));
+        BEAST_EXPECT (
+            jvSig2[jss::result][jss::status].asString() == "success");
+
+        // Save the hash with one signature for use later.
+        std::string const hash1 =
```

Comment:
Is it worth checking any of the hash characteristics here?

Reply:
Possibly?  To me the important characteristic of the hash is it that it allows me to look up the transaction in the ledger.  So the various hash checks I'm doing throughout the test show three things:
1. Adding the first signature provides us a hash.  When we add the second signature, that also provides a hash.  Those two hashes should be different.  It would be easy to say, well of course they are different.  But the ripple-online-tool had a bug where they were not different.  So it's worth checking.
2. If we submit the transaction with two signatures, the hash that comes back from that submission should be the same as the hash that we got back when we signed the second time.
3. The hash returned when we submitted the transaction (which is the same as the hash returned when we signed the second time) should be able to locate the transaction in the ledger.

Are there other characteristics I should be testing for?  Thanks.